loading-button-android/src/main/java/br/com/simplepass/loadingbutton/customViews/ProgressButton.kt

```diff
@@ -82,12 +85,20 @@ internal fun ProgressButton.init(attrs: AttributeSet? = null, defStyleAttr: Int
         getContext().obtainStyledAttributes(this, attrsArray, defStyleAttr, 0)
     }
 
-    drawable = parseGradientDrawable(
-        typedArrayBg?.getDrawable(0)
+    val tempDrawable = typedArrayBg?.getDrawable(0)
             ?: ContextCompat.getDrawable(getContext(), R.drawable.shape_default)!!
-    )
+    drawableBackground = tempDrawable.let {
+        when (it) {
+            is ColorDrawable -> GradientDrawable().apply { setColor(it.color) }
+            else -> it
+        }
+    }
+    if (typedArray?.getBoolean(R.styleable.CircularProgressButton_use_drawable_cache, false) == false) {
+        drawableBackground = drawableBackground.constantState?.newDrawable()?.mutate()
```

Comment:
changed to `let` but as mentioned above we will need to stay with Drawable because not all drawables can be casted to Gradient there is example added in test app.